### PR TITLE
Do not track associations by default

### LINKED ISSUE #740: "error while trying to load the gem" when database does not exist, or connection not yet established
Hi,
I have a fresh sinatra application on which paper_trail has been included. After I bundled it and ran
```
bundle exec rake db:create
```
It gave me error:
```
Bundler::GemRequireError: There was an error while trying to load the gem 'paper_trail'.
.........
ActiveRecord::NoDatabaseError: FATAL:  database "university_dev" does not exist
........
PG::ConnectionBad: FATAL:  database "university_dev" does not exist
```

If I use version below 4.0.0 (i.e. 3.0.9), it works fine. But after I upgrade it to 4 or above, it gives me same error as above.

### PR DESCRIPTION
**Pull request by jaredbeck**
[Fixes #740]

Alternative to https://github.com/airblade/paper_trail/pull/742

### CODE REVIEW

**Comment by josisusan** [comment]
Hi, 
In the beginning we too thought of turning it off 😄 . We didn't know the real purpose for the track_associations method so we played around the database connectivity on PR (https://github.com/airblade/paper_trail/pull/742).

**Comment by jaredbeck** [comment]
> In the beginning we too thought of turning it off ..

Thanks Susan. So you'd be OK with this PR instead of https://github.com/airblade/paper_trail/pull/742 ?

**Comment by josisusan** [comment]
@jaredbeck Yes I am Ok with this PR too.

File: lib/paper_trail/config.rb [hunk 1]
```diff
@@ -34,9 +34,19 @@ def serialized_attributes=(_)
...
 
+    # Previously, we checked `PaperTrail::VersionAssociation.table_exists?`
+    # here, but that proved to be problematic in situations when the database
+    # connection had not been established, or when the database does not exist
+    # yet (as with `rake db:create`).
     def track_associations?
       if @track_associations.nil?
-        PaperTrail::VersionAssociation.table_exists?
+        ActiveSupport::Deprecation.warn <<-EOS.strip_heredoc.gsub(/\s+/, " ")
+          PaperTrail.track_associations has not been set. As of PaperTrail 5, it
```
Comment (owenr) [inline lib/paper_trail/config.rb]: Shouldn't this read ```DEPRECATION WARNING: PaperTrail.**config**.track_associations has not been set.``` ?

Comment (jaredbeck) [inline lib/paper_trail/config.rb]: Yes, well-spotted Owen. Feel like making a PR? Thanks!

**Pull request merged**